doc/api/url.md

```diff
@@ -176,8 +176,9 @@ The formatting process operates as follows:
   * If the `urlObject.pathname` *does not start* with an ASCII forward slash
     (`/`), then the literal string '/' is appended to `result`.
   * The value of `urlObject.pathname` is appended to `result`.
-* Otherwise, if `urlObject.pathname` is not `undefined` and is not a string, an
-  [`Error`][] is thrown.
+* Otherwise, if `urlObject.pathname` is not `undefined` and is not a string, the
+  value of `urlObject.pathname` will be the result of `urlObject.pathname.toString()`
```

Comment:
This is not totally true. If value is truthy it tries to read the [`length`](https://github.com/nodejs/node/blob/c184f22997f5ade1201ad17ebc82acc5cdac9da6/lib/url.js#L592) property. If there isn't a length property or if its value is falsy then `pathname` is converted to a string when added to the result, but

```js
url.format({ pathname: Buffer.from('a') });
```

throws.